packages/hint-manifest-icons/src/hint.ts

```diff
@@ -267,6 +270,43 @@ export default class ManifestIconHint implements IHint {
             return validSizes;
         };
 
+        const validateIconsPurpose = (icons: ManifestImageResource[], resource: string, getLocation: JSONLocationFunction) => {
+            for (let iconIndex = 0; iconIndex < icons.length; iconIndex++) {
+                const icon = icons[iconIndex];
+
+                if (!icon.purpose) {
+                    continue;
+                }
+
+                const purposes = icon.purpose.split(' ').filter((purpose) => {
+                    return purpose;
+                });
```

Comment:
Is this `filter` to remove any possible trailing space? For example, if the purpose is `maskable badge `?
If so I think it will be more readable if you just do this:

```suggestion
                const purposes = icon.purpose.trim().split(' ');
```

Reply:
It is also to remove possible extra spaces in the middle of the string: `any       badge`

Reply:
The following should work but at this time I'm not sure what's easier to read

```suggestion
                const purposes = icon.purpose.trim().replace(/\s\s*/g,' ').split(' ').;
```

@johnemau what do you think?

Reply:
for readability, I think the regex is easy and more clear what we are doing. Performance wise, I'm not sure what is better.

Reply:
Let's optimize for readability then, I don't think we can really measure the impact of one or the other 😊

Reply:
done